src/Xrefcheck/System.hs

```diff
@@ -51,125 +44,122 @@ askWithinCI = lookupEnv "CI" <&> \case
   Just (map C.toLower -> "true") -> True
   _ -> False
 
--- | A FilePath that has been canonicalized.
+-- | Relative file path with POSIX path separators.
 --
--- It should be created via 'canonicalizePath'.
+-- This type exist in contrast to 'FilePath' which, in this project,
+-- is used for platform-dependent file paths and related filesystem
+-- IO operations.
+newtype RelPosixLink = RelPosixLink
```

Comment:
If `RelPosixLink` is safe constructor, I would then mention explicitly that non-posix separators can also appear here and that they will serve as mere part of filename. And that this is the logic that GitHub/GitLab follow.